utility/src/main/java/org/oppia/util/networking/NetworkConnectionUtil.kt

```diff
@@ -0,0 +1,40 @@
+package org.oppia.util.networking
+
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.NetworkInfo
+import androidx.annotation.VisibleForTesting
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class NetworkConnectionUtil @Inject constructor(private val context: Context) {
+  enum class ConnectionStatus {
+    WIFI, CELLULAR, NONE
+  }
+  private var connectionStatus: ConnectionStatus? = null
+
+  fun getCurrentConnectionStatus(): ConnectionStatus {
+    connectionStatus?.let {
+      return it
+    }
+    val connectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+    val activeNetwork: NetworkInfo? = connectivityManager.activeNetworkInfo
+    val isConnected: Boolean = activeNetwork?.isConnected == true
+    val isWiFi: Boolean = activeNetwork?.type == ConnectivityManager.TYPE_WIFI
+    val isMob: Boolean = activeNetwork?.type == ConnectivityManager.TYPE_MOBILE
+    if(isConnected) {
```

Comment:
Nit: spacing between ``if`` and ``(``. Also, I think we can simplify this a bit with a ``when`` statement:

```kotlin
val connectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
return connectivityManager.activeNetworkInfo?.let { activeNetwork ->
  val isConnected = activeNetwork.isConnected
  val isLocal = activeNetwork.type == ConnectivityManager.TYPE_WIFI || ConnectivityManager.TYPE_ETHERNET
  val isCellular = activeNetwork.type == ConnectivityManager.TYPE_MOBILE || ConnectivityManager.TYPE_WIMAX
  return@let when {
    isConnected && isLocal -> ConnectionStatus.LOCAL
    isConnected && isCellular -> ConnectionStatus.CELLULAR
    else -> ConnectionStatus.NONE
  }
} ?: ConnectionStatus.NONE
```

I also rearranged some things and renamed WIFI to LOCAL since Ethernet is definitely a valid transport type for streaming audio.

Reply:
Updated with this code block.